src/dune_pkg/archive_driver.ml

```diff
@@ -26,38 +26,50 @@ let make_zip_args ~archive ~target_in_temp =
 
 let tar =
   let command =
-    lazy
-      (match
+    Fiber.Lazy.create
+      (fun () -> match
          (* Test for tar before bsdtar as tar is more likely to be installed
             and both work equally well for tarballs. *)
          List.find_map [ "tar"; "bsdtar" ] ~f:which
        with
-       | Some bin -> { Command.bin; make_args = make_tar_args }
+       | Some bin -> Fiber.return { Command.bin; make_args = make_tar_args }
        | None -> Dune_engine.Utils.program_not_found "tar" ~loc:None)
   in
   { command; suffixes = [ ".tar"; ".tar.gz"; ".tgz"; ".tar.bz2"; ".tbz" ] }
 ;;
 
+let which_bsdtar (bin_name : string) =
+  let bin_path =
+    match which bin_name with
+    | Some bin -> bin
+    | None -> Dune_engine.Utils.program_not_found "unzip" ~loc:None
```

Comment:
This is probably wrong, if it searches for `tar` or `bsdtar` it shouldn't fail with an error that `unzip` can't be found.

```suggestion
    | None -> Dune_engine.Utils.program_not_found bin_name ~loc:None
```

Reply:
Hm, I'm not so sure about this. The reason we're looking for `bsdtar` or `tar` is because `unzip` doesn't exist, and `unzip` is the definitive program to extract zip files, while `tar` may or may not extract zip files. I think the error message should be clear here that the failure is because a zip file is not being extracted. Should we write a custom error message here?